### DISCUSSION TITLE
Blank web page with fresh install with Docker.

### DISCUSSION THREAD

**Discussion by StevenIsaacs** [inventree/InvenTree#8459] [Q&A] [answered]
Performed fresh install using the instructions on: https://docs.inventree.org/en/latest/start/docker_install

Attempting to connect to the server using Chrome results in a blank white page. The debug window in Chrome contains:
```
<html> == $0
  <head></head>
  <body></body>
</html>
```
Curl produces:
```
curl http://inventory -v
* Host inventory:80 was resolved.
* IPv6: (none)
* IPv4: 172.31.0.75
*   Trying 172.31.0.75:80...
* Connected to inventory (172.31.0.75) port 80
> GET / HTTP/1.1
> Host: inventory
> User-Agent: curl/8.6.0
> Accept: */*
> 
< HTTP/1.1 200 OK
< Server: Caddy
< Date: Fri, 08 Nov 2024 13:56:30 GMT
< Content-Length: 0
< 
* Connection #0 to host inventory left intact
```
The command `docker logs "inventree-proxy"` contains:
```
{"level":"info","ts":1731073141.4802504,"logger":"http.log.access","msg":"NOP","request":{"remote_ip":"172.31.0.75","remote_port":"60902","client_ip":"172.31.0.75","proto":"HTTP/1.1","method":"GET","host":"inventory","uri":"/","headers":{"User-Agent":["curl/8.5.0"],"Accept":["*/*"]}},"bytes_read":0,"user_id":"","duration":0.000002719,"size":0,"status":0,"resp_headers":{"Server":["Caddy"]}}
{"level":"info","ts":1731074190.1960294,"logger":"http.log.access","msg":"NOP","request":{"remote_ip":"172.31.0.77","remote_port":"33450","client_ip":"172.31.0.77","proto":"HTTP/1.1","method":"GET","host":"inventory","uri":"/","headers":{"User-Agent":["curl/8.6.0"],"Accept":["*/*"]}},"bytes_read":0,"user_id":"","duration":0.000007546,"size":0,"status":0,"resp_headers":{"Server":["Caddy"]}}
```

The `"msg":"NOP"` doesn't look right to me.

In the `.env` file:
 * INVENTREE_EXT_VOLUME=./pfs-inventory
 * INVENTREE_DB_USER=pfsinventory
 * INVENTREE_DB_PASSWORD=XXXXXXX
 * INVENTREE_SITE_URL="http://172.31.0.75"

The pfs-inventory directory was created and appears to be fully populated.

The `Caddyfile` was changed:
```
# Change the host to your domain (this will serve at inventree.localhost)
{$INVENTREE_SITE_URL:172.31.0.75} {
```

The server is running:
```
lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 24.04 LTS
Release:	24.04
Codename:	noble
```

Any suggestions?

BTW: Attempts to install on Fedora 40 is a series of failures starting with no `docker compose` available.

Thanks,
Steve

**Comment by SchrodingersGat** [comment, marked as answer]
> curl http://inventory -v

> INVENTREE_SITE_URL="http://172.31.0.75/"

The Caddy proxy is only expecting to service 172.31.0.75 - change your `INVENTREE_SITE_URL` to `http://inventory` (and same in your Caddyfile).

> BTW: Attempts to install on Fedora 40 is a series of failures starting with no docker compose available.

From [the documentation](https://docs.inventree.org/en/stable/start/docker_install/#docker-production-server):

![image](https://github.com/user-attachments/assets/f40c1dc1-a402-4247-8634-25689b445947)

**Reply by StevenIsaacs** [reply]
As confirmed by the curl output http://inventory resolves to 172.31.0.75. This is by virtue of the /etc/hosts file. Using http://inventory did not work either.

I upgraded the Fedora machine to 41 and now docker has the compose command. I then did the install on Fedora 41 and was able to bring up the server. However, when connecting using  Chrome I get exactly the same results as before -- a blank page.

**Reply by matmair** [reply]
You need to set site_url to the exact domain or ip you want to use in your browser to access the site. Both caddy and the server ignore any other traffic.

**Reply by StevenIsaacs** [reply]
Ok, thanks for the clue. Using the IP address instead of the http://inventory URL in my browser presents me with a login page. 

First the server is on the local network and a DNS server is not running on the local network. Instead, I was relying on /etc/hosts to configure a name for an IP thus: http://inventory. My attempts at using that URL in .env and Caddyfile resulted in failures while initializing the containers. I then changed the configs to use the IP address instead which got me past the initialization failures. Thinking that /etc/hosts would resolve correctly I then used the http://inventory URL in the browser because the /etc/hosts would resolve the IP address. Using curl to me confirmed this would work -- which apparently it doesn't.

There is some inconsistency in name resolution I don't understand and won't dig into any further. I'm at a login page so have more pressing things to do.

Thanks to both of you for helping me get off of this rock in the road.

Steve

**Reply by matmair** [reply]
Name resolution is probably not the problem here. InvenTree only responds with the application to the exact address (be it hostname or IP) configured. This is a security measure against a host of MITM and confusion attacks.
